tools/azsdk-cli/docs/specs/6-package-updater.spec.md

```diff
@@ -0,0 +1,526 @@
+
+  style azsdk_cli fill:#e3f2fd,stroke:#0277bd
+  style Agent fill:#e1f5fe,stroke:#0288d1
+```
+
+---
+
+### Workflow Diagram
+
+#### Update-Changelog (mgmt-plane)
+
+```mermaid
+flowchart TD
+      A[Start: UpdateChangelogAsync] --> B[Validate packagePath] 
+      B --> C{packagePath valid?}
+      C -->|No| D[Return failure: Package path required/invalid]
+      C -->|Yes| E[Discover SDK repository root] 
+    
+      E --> F{Repository root found?}
+      F -->|No| G[Return failure: Failed to discover SDK repo]
+      F -->|Yes| H[Get update-changelog configuration] 
+    
+      H --> I{Configuration retrieval successful?}
+      I -->|No| J[Return failure: Failed to get update-changelog configuration] 
+      I -->|Yes| K{Configuration type?}
+    
+      K -->|Script| L[Run update-changelog script]
+      K -->|None| M[Run CLI built-in code] 
+    
+      L --> N[Prepare result] 
+      M --> N
+    
+      N --> O{update-changelog process completed?} 
+      O -->|Failed| P[Return failure with exit code and output]
+      O -->|Success| Q[Return success]
+    
+      style A fill:#e1f5fe
+      style Q fill:#c8e6c9
+      style D fill:#ffcdd2
+      style G fill:#ffcdd2
+      style J fill:#ffcdd2
+      style P fill:#ffcdd2
+```
+
+> **Note:** Other tools (version update, metadata update, CI update) follow a similar flowchart pattern.
+
+---
+
+## Open Questions
+
+---
+
+## Success Criteria
+
+This tool set is complete when:
+
+- [ ] All four tools emit structured JSON with consistent schema.
+- [ ] Management-plane package run updates version + changelog automatically.
+      Exception: Java returns `no-op` for `update-version` tool.
+- [ ] Data-plane run returns advisory for changelog and version.
+- [ ] Metadata files are updated accordingly.
+
+    | Language | Exception |
+    |---|---|
+    |Java|No additional files need updating|
+    |JavaScript|No additional files need updating|
+    |Go|No additional files need updating|
+
+- [ ] Cross-language scripts execute without manual intervention.
+- [ ] Errors surface actionable mitigation text.
+
+---
+
+## Agent Prompts
+
+### Scenario 1: Update Management-Plane Package (existing SDK) for Release
+
+**Prompt:**
+
+```text
+Update the package at /home/dev/sdk/healthdataaiservices/Azure.ResourceManager.HealthDataAIServices for a stable release.
+```
+
+**Expected Agent Activity:**
+
+1. Run changelog update.
+2. Run version update with inferred `--release-type stable`.
+3. Run metadata update (README, API export, formatting if needed) and summarize JSON results.
+
+### Scenario 2: Update Version for Stable Release (Data-Plane)
+
+**Prompt:**
+
+```text
+Update the version and release date for package at /repo/sdk/healthdataaiservices/Azure.Health.Deidentification.
+```
+
+**Expected Agent Activity:**
+
+1. Confirm with user on the version to use.
+2. Run version update with the confirmed version.
+3. Report the result.
+
+### Scenario 3: Create CI Config for New SDK
```

Comment:
I dont see this CI update tool outlined above - is this something already in the azsdkcli?

Reply:
No, they were in the spec before. I removed them from the spec based on the feedback since it won't be included in milestone 1.